contracts/reflect/src/query.rs

```diff
@@ -0,0 +1,96 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{
+    from_slice, to_binary, Binary, Querier, QuerierResult, QueryRequest, StdResult, SystemError,
+};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+/// CustomQuery is an override of QueryRequest::Custom to show this works and can be extended in the contract
+pub enum CustomQuery {
+    Ping {},
+    Capital { text: String },
+}
+
+// TODO: do we want to standardize this somehow for all?
+impl Into<QueryRequest<CustomQuery>> for CustomQuery {
+    fn into(self) -> QueryRequest<CustomQuery> {
+        QueryRequest::Custom(self)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+// All return values of CustomQuery are CustomResponse
+pub struct CustomResponse {
+    pub msg: String,
+}
+
+impl CustomQuery {
+    fn execute(&self) -> StdResult<Binary> {
+        let msg = match self {
+            CustomQuery::Ping {} => "pong".to_string(),
+            CustomQuery::Capital { text } => text.to_uppercase(),
+        };
+        to_binary(&CustomResponse { msg })
+    }
+}
+
+#[derive(Clone)]
+pub struct CustomQuerier {}
```

Comment:
Do I understand correctly, that `CustomQuerier` is used in test code only, since it is a type that lives in the runtime, not the contract? If so, why don't we get unused warnings when running `cargo wasm`?

Anyways, I think we should somehow structure code to make clear that is is not included in the contract logic. Which is hard if the code should be reused in unit tests and integration tests. [We faced this issue before](https://github.com/CosmWasm/cosmwasm-examples/issues/57): where to put custom non-contract test code.

Reply:
Yes,this is just for test code to simulate a runtime, like MockQuerier.

Maybe we making a testing module?

We don't get compiler warnings as we export it via pub mod query, which means we expect externals to use it (integration test) and thus no unused module warning.

Reply:
Yeah, a

```rust
#[cfg(not(target_arch = "wasm32"))]
pub mod testing;
```

in `lib.rs` would make it much easier to follow how the different components fit together.

Reply:
Done